internal/integration/fetcher.go

```diff
@@ -176,7 +181,8 @@ func (pf *prometheusFetcher) Fetch(targets []endpoints.Target) <-chan TargetMetr
 				Info("Target list for fetching metrics is empty")
 			return
 		}
-		ticker := time.NewTicker(pf.duration / time.Duration(nTargets))
+
+		ticker := time.NewTicker((pf.duration / 2) / time.Duration(nTargets))
```

Comment:
why half of the duration?

Reply:
I just pushed a new commit explaining why!